explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+ * mapping of host names to IP addresses, and vice-versa
+ * information about other services, such as kerberos realms, LDAP servers, etc (usually on internal domains only)
+ * and more
+
+When DNS was first conceived, security wasn't a top priority yet. At its origins, DNS is susceptible to multiple vulnerabilities, and has many weaknesses. Most of them are a consequence of spoofing: there is no guarantee that the reply you got to a DNS query a) was not tampered with; b) came from the true source.
+
+This is not news, and other mechanisms on top of DNS and around it are in place to counteract that weakness. For example, the famous *HTTPS* padlock that can be seen when accessing most websites nowadays, which uses the TLS protocol to both authenticate the website, and encrypt the connection. It doesn't prevent DNS spoofing, and your web browser might still be tricked into attempting a connection with a fraudulent website, but the moment the TLS certificate is inspected, a warning will be issued to the user. Depending on local policies, the connection might be even immediately blocked. Still, DNS spoofing is a real problem, and TLS itself is subject to other types of attacks.
+
+## What is it?
+
+DNSSEC stands for Domain Name System (DNS) Security Extensions (SEC). It's an extension to DNS that introduces digital signatures so that each DNS response can be verified for:
+
+ * integrity: The answer was not tampered with and did not change during transit.
+ * authenticity: The data came from the true source, and not another entity impersonating the source.
+
+It's important to note that DNSSEC, however, will NOT encrypt the data: it is still sent in the clear.
+
+DNSSEC is based on public key cryptography, meaning that every DNS zone has a public/private key pair. The private key is used to sign the zone's DNS records, and the corresponding public key can be used to verify those signatures. This public key is also published in the zone, and anyone fetching records can also fetch the public key to verify the signature of the data.
+
+The question then becomes, how to trust that this public key is authentic? Turns out the key is also signed: it's signed by the parent zone's key, which is also signed by its parent, and so on, all the way to the top: the root DNS zone. The root zone keys are trusted implicitly, and all DNS resolvers have it as a trust anchor. This sequence of keys and signatures all the way to the top is called the chain of trust.
+
+The public key cryptography behind SSL/TLS is similar, but there we have the Certificate Authority entity (CA) that issues the certificates and vouches for them. Every single web browser or other SSL/TLS client or operating system needs to have a "bootstrap" list of CAs that it will trust by default, and there are dozens. In DNSSEC, the only bootstrap public key the resolver needs is the root zone one. It's as if there were only one trusted CA.
```

Comment:
"The public key cryptography underpinning SSL/TLS operates in a similar manner, but relies on Certificate Authority (CA) entities to issue and vouch for certificates. Web browsers and other SSL/TLS clients must maintain a list of trusted CAs, typically numbering in the dozens. In contrast, DNSSEC simplifies this process by requiring only a single trusted root zone key. This streamlined approach effectively reduces the complexity of establishing trust."